TimerTriggeredCalcTests/UnitTests.cs

```diff
@@ -8,118 +8,111 @@
+            const int CancellationThreshold = 15; // canceling the timer object is slightly unpredictable, don't fail the test unless it loops far too many times
 
-            var tempValToWrite = 273.0;
-            var pressValToWrite = 2280.0;
-            var volValue = 500;
-            var gasConstant = 62.363598221529; // units of  L * Torr / (K * mol)
-            var expectedMolesOutput = pressValToWrite * volValue / (gasConstant * tempValToWrite);
-            var contextElementList = new List<AFElement>();
-            var templateName = "EventTriggeredSampleTemplate";
+            const double TemperatureValueToWrite = 273.0;
+            const double PressureValueToWrite = 2280.0;
+            const int VolumeValue = 500;
+            const double GasConstant = 62.363598221529; // units of  L * Torr / (K * mol)
+            const double ExpectedMolesOutput = PressureValueToWrite * VolumeValue / (GasConstant * TemperatureValueToWrite);
+            List<AFElement> contextElementList = new List<AFElement>();
+            const string TemplateName = "EventTriggeredSampleTemplate";
 
             AFDatabase myAFDB = null;
 
             try
             {
                 #region configurationSettings
-                string solutionFolderName = Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.FullName;
+                string solutionFolderName = Directory.GetParent(Directory.GetCurrentDirectory())?.Parent?.Parent?.FullName;
                 AppSettings settings = JsonSerializer.Deserialize<AppSettings>(File.ReadAllText(solutionFolderName + "/TimerTriggeredCalc/appsettings.json"));
+
+                if (settings == null) throw new FileNotFoundException("Could not find appsettings.json file");
                 #endregion // configurationSettings
 
                 #region step1
-                Console.WriteLine("TEST: Resolving AF Server object...");
+                _testOutputHelper.WriteLine("TEST: Resolving AF Server object...");
 
-                var myPISystems = new PISystems();
+                PISystems myPISystems = new PISystems();
                 PISystem myPISystem;
 
-                if (string.IsNullOrWhiteSpace(settings.AFServerName))
-                {
-                    // Use the default PI Data Archive
-                    myPISystem = myPISystems.DefaultPISystem;
-                }
-                else
-                {
-                    myPISystem = myPISystems[settings.AFServerName];
-                }
+                myPISystem = string.IsNullOrWhiteSpace(settings.AFServerName) ? myPISystems.DefaultPISystem : myPISystems[settings.AFServerName];
                 
                 if (myPISystem is null)
                 {
-                    Console.WriteLine("Create entry for AF Server...");
+                    _testOutputHelper.WriteLine("Create entry for AF Server...");
```

Comment:
I'm suprised these don't trigger a naming violation. It's nitpicky, but if we were strict these would be PascalCase after the underscore instead of camelCase